### DISCUSSION TITLE
Recommended way to codegen across separate repos

### DISCUSSION THREAD

**Discussion by hobbsh** [dotansimha/graphql-code-generator#8203] [Q&A] [answered]
Hi,

We have services split up into separate repos and I'm wondering if there's a way to do generation within one repo (where our frontend lives) using a GraphQL schema that lives in another, apart from the obvious (cloning + relative pathing)

**Comment by charlypoly** [comment, marked as answer]
Hi @hobbsh!

Apart from cloning the repo or using relative pathing, there are many ways to fetch the schema with codegen:
1. another obvious solution is to point directly to your server URL so that codegen can perform an introspection query
2. however, if you disabled introspection in production (or staging), you can also use a Schema Registry that will host it for you. Aside from codegen, we are also maintaining dozen of other GraphQL projects, including an open-source Schema Registry service: [GraphQL Hive](https://graphql-hive.com/). The free tier is sufficient for most users, and you can also host it yourself.
3. You can also pass a [Github URL](https://www.graphql-code-generator.com/docs/config-reference/schema-field#github) of the schema file along with some authentication headers (that would need to be configured in local env)

Let me know if I can provide more details on one of the proposed solution.

**Reply by hobbsh** [reply]
Thanks for the quick reply @charlypoly! Currently we are just exposing our subgraph like you mentioned in point 1, but it's not ideal. We did look at the GitHub integration but that seemed like the most cumbersome at the moment but will definitely keep it in the back pocket. 

We are currently using Apollo Studio for our schema registry and while we could use `rover subgraph fetch` to grab the schema, I'm wondering if `codegen` can integrate directly with it? Would it basically be [this](https://www.graphql-code-generator.com/docs/config-reference/schema-field#apollo-engine)?

**Reply by charlypoly** [reply]
If you are using Apollo Studio, the last mentioned option (using `apollo-client` option) is the proper one.

**Reply by hobbsh** [reply]
@charlypoly I'm assuming you meant `apollo-engine` not `apollo-client`? If so, I don't see a way to target a specific subgraph this way, which is ultimately our issue. Doing it this way we would be targeting a supergraph so if we make a conflicting change locally, `codegen` will not work. What do you recommend?

**Reply by charlypoly** [reply]
I meant `apollo-engine`, but actually, after checking, it does not support Apollo Studio.

Apart from the aforementioned mentioned 3 solutions, another one would be to write your own loader that will fetch the proper schema: https://www.graphql-code-generator.com/docs/config-reference/schema-field#loading-api-url-from-typescript-file-example

**Reply by hobbsh** [reply]
Thanks, we tried it with Apollo Studio confirmed it does not work. We ended up using `rover subgraph fetch` for now as it integrated easiest into our flow, but it would be great to integrate `codgen` directly with Apollo Studio at some point.
